### ISSUE TITLE
ReplayStateMachine tries to complete recovery future multiple times

### ISSUE THREAD

**Issue by npepinpe** [camunda/zeebe#11683]
**Describe the bug**

An error occurred in SaaS where the `ReplayStateMachine` tried completing the recovery future multiple times. It completed it successfully once, then tried to complete it a second time successfully (unclear if with the same values, the message doesn't say). This threw an error, at which point the catch block tried to complete the future exceptionally (:smile:), which of course also failed.

The node was already shutting down however, so the impact was minimal in this case. If this is because the node was shutting down or not is unclear.

By the way, it looks to me like if there is ever any error in the replay state machine (even if we're constantly replaying), we always try to complete the recovery future exceptionally? This could trigger errors in general.

Error link: https://console.cloud.google.com/errors/detail/CNr7q4KVnLXBXg;service=zeebe;time=P7D?project=camunda-cloud-240911

**To Reproduce**

No obvious means to do so in the logs or code at first glance.

**Expected behavior**

The recovery future is completed only once, or we remove that constraint from our `CompletableActorFuture` instances.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.IllegalStateException: Cannot complete future, the future is already completed  with value [lastProcessedPosition: 86753, lastWrittenPosition: 86756]
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.completeExceptionally(CompletableActorFuture.java:186) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.completeExceptionally(CompletableActorFuture.java:193) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.replayNextEvent(ReplayStateMachine.java:193) ~[zeebe-stream-platform-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:92) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
Caused by: java.lang.RuntimeException: Failed to replay records. [snapshot-position: 86753, last-read-record-position: 86756, last-replayed-event-position: -1]
	... 7 more
Caused by: java.lang.IllegalStateException: Cannot complete future, the future is already completed  with value [lastProcessedPosition: 86753, lastWrittenPosition: 86756]
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.complete(CompletableActorFuture.java:166) ~[zeebe-scheduler-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.onRecordsReplayed(ReplayStateMachine.java:256) ~[zeebe-stream-platform-8.2.0-alpha4.jar:8.2.0-alpha4]
	at io.camunda.zeebe.stream.impl.ReplayStateMachine.replayNextEvent(ReplayStateMachine.java:182) ~[zeebe-stream-platform-8.2.0-alpha4.jar:8.2.0-alpha4]
	... 6 more 
```

</p>
</details>

**Environment:**
- Zeebe Version: 8.2.0-alpha4
- Configuration: SaaS

**Comment by oleschoenburg** [comment]
See also https://github.com/camunda/zeebe/issues/11060. Not the first time we've had this :eyes:

**Comment by npepinpe** [comment]
Shouldn't it have been fixed then? :thinking:

**Comment by megglos** [comment]
planning: it's mostly noise, however this technically shouldn't happen, keep in backlog for now

**Comment by remcowesterhoud** [comment]
Had another occurrence of this in `8.3.0-alpha1`

**Comment by Zelldon** [comment]
Q: Isn't this a duplicate of https://github.com/camunda/zeebe/issues/8528

**Comment by npepinpe** [comment]
Could be, it's been too long. Feel free to close it in favor of the other :shrug:

**Comment by Zelldon** [comment]
Ok I will close it :)
